app/helpers/trainee_helper.rb

```diff
@@ -34,4 +34,15 @@ def trainees_page_title(trainees, total_trainees_count)
       total_trainees_count_text: total_trainees_count_text,
     )
   end
+
+  def show_publish_courses?(trainee)
+    courses_available = trainee.available_courses.present?
+    manual_entry_chosen = PublishCourseDetailsForm.new(trainee).manual_entry_chosen?
+
+    FeatureService.enabled?(:publish_course_details) && courses_available && !manual_entry_chosen
+  end
+
+  def stashed_code(trainee)
+    PublishCourseDetailsForm.new(trainee).code || "not_used"
```

Comment:
Should this be in the helper?

Feels like `PublishCourseDetailsForm.new(trainee).code` should return `"not_used"`

Reply:
This is a bit weird. I put it here because this line https://github.com/DFE-Digital/register-trainee-teachers/pull/731/files#diff-b7b818e8f3b3ef5e954af2cc85c9a33fae2d114228f0006b1f2a80d3c872d130R83 needs *something* (not nil) to generate a path, except in the situation that we dont have a code, we dont end up showing the user that path anyway. I thought it would be surprising to have the form return "not_used" when the value is actually nil, and this encapsulates the weirdness into this one helper method. 

I considered adding a helper method to generate the route to make this clearer eg:
```
 def publish_details_confirm_path(@trainee)
code = PublishCourseDetailsForm.new(trainee).code
return unless code # User will not be shown this confirm page if there is no code

 edit_trainee_confirm_publish_course_path(id: PublishCourseDetailsForm.new(trainee).code, trainee_id: @trainee.to_param)
 end
```

Do you think that is better?

Reply:
another alternative is to allow the `path` and `confirm_path` values to be lambdas in `Tasklist::View::Row`